wave_calculator.gd

```diff
@@ -6,19 +6,50 @@ var rng: RandomNumberGenerator = RandomNumberGenerator.new()
 var _base_spawn_time: float
 var _enemies: Dictionary  # Was not able to type the dict either
 
+const MAX_WAVE_BATCH_SCORE = 10
+
 
 func _init(base_spawn_time: float, enemies: Dictionary) -> void:
 	_base_spawn_time = base_spawn_time
 	_enemies = enemies
 
 
 # Array[Dictionary] doesn't work :sad: https://forum.godotengine.org/t/array-with-dictionaries-doesnt-get-recognized-as-array-dictionary/58280
-func get_enemy_spawns(wave_number: int) -> Array:
+func get_enemy_spawns(wave_number: int):
 	var difficulty = _get_difficulty(wave_number)
+	var total_batches = max(1, ceil(float(difficulty) / MAX_WAVE_BATCH_SCORE))
```

Comment:
We can try it out like this, but when the difficulty gets high enough, the wave will be rather long instead of difficult, particularly because we have a fixed wait between batches.

Reply:
add variable wave batch frequency that's based on difficulty (completely made up the formula) :man_shrugging: